Caboodle/Geocoding/Geocoding.uwp.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Windows.Devices.Geolocation;
+using Windows.Services.Maps;
+
+namespace Microsoft.Caboodle
+{
+	public partial class Geocoding
+    {
+        public static async Task<IEnumerable<Placemark>> GetPlacemarksAsync(double latitude, double longitude)
+        {
+
+            var queryResults =
+                await MapLocationFinder.FindLocationsAtAsync(
+                        new Geopoint(new BasicGeoposition { Latitude = latitude, Longitude = longitude })).AsTask();
+
+            return queryResults?.Locations.ToPlacemarks();
+        }
+
+        public static async Task<IEnumerable<Location>> GetLocationsAsync(string address)
+        {
+            SetMapKey();
+
+            var queryResults = await MapLocationFinder.FindLocationsAsync(address, null, 10);
+
+            if (queryResults?.Locations == null)
+                return null;
+
+            return queryResults.Locations.Select(l  => new Location
+			{
+				Latitude = l.Point.Position.Latitude,
+				Longitude = l.Point.Position.Longitude
+			});
+        }
+
+        static void SetMapKey()
```

Comment:
If we do go the way of platform properties on each API, then we should call this something like: `EnsureMapKeyValid`.

Reply:
Probably ValidateMapKey? It is internal